packages/grafana-ui/src/components/VizLayout/VizLayout.story.tsx

```diff
@@ -9,19 +9,26 @@ export default {
   decorators: [withCenteredStory],
   parameters: {
     docs: {},
+    knobs: {
+      disable: true,
+    },
+    controls: {
+      exclude: ['legend'],
+    },
+  },
+  argTypes: {
+    width: { control: { type: 'range', min: 100, max: 1000 } },
+    height: { control: { type: 'range', min: 100, max: 1000 } },
+    legendWidth: { control: { type: 'range', min: 100, max: 280 } },
+    legendItems: { control: { type: 'number', min: 1 } },
   },
 };
 
-const getKnobs = () => {
-  return {
-    legendWidth: number('legendWidth', 100),
-    legendItems: number('legendItems', 2),
-  };
-};
-
-export const BottomLegend = () => {
-  const { legendItems } = getKnobs();
-  const items = Array.from({ length: legendItems }, (_, i) => i + 1);
+export const BottomLegend: Story = ({ height, width, legendItems }) => {
+  const [items, setItems] = useState(Array.from({ length: legendItems }, (_, i) => i + 1));
```

Comment:
The array creating logic could probably be abstracted into a separate function, as it's used in more than one place.

Reply:
alright 👍
Would fix up right away